AsyncDisplayKit/ASCellNode.h

```diff
@@ -14,6 +14,15 @@ NS_ASSUME_NONNULL_BEGIN
 
 typedef NSUInteger ASCellNodeAnimation;
 
+typedef enum : NSUInteger {
+  // Indicates a cell has just became visible
+  ASCellNodeVisibilityEventVisible,
```

Comment:
Is there any value to having symmetric `Visible`/`Invisible` events.  I suspect it will be a fairly common need for callers to do something at the point where a cell transitions from invisible to visible.  In the Pinterest case, that action would be "save the current time as the start of a new impression".  Currently, with `EventVisibleRectChanged`, we'll have to store some BOOL `isVisible` state that we use inside of `EventVisibleRectChanged` to detect that transition between invisible and visible.